File: prompt-enhancer-extension/generate-icons.py
#!/usr/bin/env python3
# Generate PNG icons for Chrome extension

from PIL import Image, ImageDraw, ImageFont
import os

# Create icons directory if it doesn't exist
icons_dir = 'icons'
if not os.path.exists(icons_dir):
    os.makedirs(icons_dir)

def create_icon(size):
    # Create image with gradient-like background
    img = Image.new('RGB', (size, size), color='#764ba2')
    draw = ImageDraw.Draw(img)
    
    # Draw rounded rectangle background
    draw.rectangle([0, 0, size, size], fill='#667eea')
    
    # Add a simple gradient effect
    for i in range(size):
        color_value = int(102 + (118-102) * i / size)  # Gradient from #667eea to #764ba2
        draw.line([(i, 0), (i, size)], fill=(color_value, 126, 234 - int(114 * i / size)))
    
    # Add sparkles emoji or text
    try:
        # Try to get a font
        font_size = int(size * 0.6)
        font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", font_size)
    except:
        # Use default font if truetype not available
        font = ImageFont.load_default()
    
    # Draw text (sparkles emoji or "PE")
    text = "✨"
    try:
        bbox = draw.textbbox((0, 0), text, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]
    except:
        text_width = size * 0.6
        text_height = size * 0.6
    
    x = (size - text_width) / 2
    y = (size - text_height) / 2
    
    try:
        draw.text((x, y), text, fill='white', font=font)
    except:
        # Fallback to "PE" if emoji fails
        text = "PE"
        draw.text((size * 0.2, size * 0.3), text, fill='white', font=font)
    
    # Save PNG
    output_path = os.path.join(icons_dir, f'icon{size}.png')
    img.save(output_path, 'PNG')
    print(f'✓ Created {output_path}')

# Generate icons
sizes = [16, 48, 128]
for size in sizes:
    create_icon(size)

print('\n✓ All PNG icons generated successfully!')

File: prompt-enhancer-extension/generate-icons.js
#!/usr/bin/env node
// Simple icon generator for Chrome extension
// Generates placeholder icons in PNG format

const fs = require('fs');
const path = require('path');

// Create icons directory
const iconsDir = path.join(__dirname, 'icons');
if (!fs.existsSync(iconsDir)) {
  fs.mkdirSync(iconsDir);
}

// Function to generate SVG icon
function generateSVG(size) {
  const svg = `
<svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="grad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#667eea;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#764ba2;stop-opacity:1" />
    </linearGradient>
  </defs>
  <rect width="${size}" height="${size}" rx="${size * 0.2}" fill="url(#grad)"/>
  <text x="50%" y="50%" font-family="Arial, sans-serif" font-size="${size * 0.6}" font-weight="bold" fill="white" text-anchor="middle" dominant-baseline="middle">✨</text>
</svg>`;
  return svg;
}

// Generate icons for different sizes
const sizes = [16, 48, 128];

sizes.forEach(size => {
  const svg = generateSVG(size);
  const svgPath = path.join(iconsDir, `icon${size}.svg`);
  fs.writeFileSync(svgPath, svg);
  console.log(`✓ Generated icon${size}.svg`);
});

console.log('\nIcons generated successfully!');
console.log('\nTo convert SVG to PNG (required for Chrome):');
console.log('Option 1: Use online converter like https://cloudconvert.com/svg-to-png');
console.log('Option 2: Use ImageMagick:');
console.log('  convert icons/icon16.svg icons/icon16.png');
console.log('  convert icons/icon48.svg icons/icon48.png');
console.log('  convert icons/icon128.svg icons/icon128.png');
console.log('\nOption 3: Use this script with canvas:');
console.log('  npm install canvas');
console.log('  node generate-icons.js --convert');

// If --convert flag and canvas is available
if (process.argv.includes('--convert')) {
  try {
    const { createCanvas } = require('canvas');
    
    sizes.forEach(size => {
      const canvas = createCanvas(size, size);
      const ctx = canvas.getContext('2d');
      
      // Create gradient
      const gradient = ctx.createLinearGradient(0, 0, size, size);
      gradient.addColorStop(0, '#667eea');
      gradient.addColorStop(1, '#764ba2');
      
      // Draw rounded rectangle
      const radius = size * 0.2;
      ctx.beginPath();
      ctx.moveTo(radius, 0);
      ctx.lineTo(size - radius, 0);
      ctx.quadraticCurveTo(size, 0, size, radius);
      ctx.lineTo(size, size - radius);
      ctx.quadraticCurveTo(size, size, size - radius, size);
      ctx.lineTo(radius, size);
      ctx.quadraticCurveTo(0, size, 0, size - radius);
      ctx.lineTo(0, radius);
      ctx.quadraticCurveTo(0, 0, radius, 0);
      ctx.closePath();
      ctx.fillStyle = gradient;
      ctx.fill();
      
      // Draw emoji
      ctx.font = `bold ${size * 0.6}px Arial`;
      ctx.fillStyle = 'white';
      ctx.textAlign = 'center';
      ctx.textBaseline = 'middle';
      ctx.fillText('✨', size / 2, size / 2);
      
      // Save PNG
      const buffer = canvas.toBuffer('image/png');
      const pngPath = path.join(iconsDir, `icon${size}.png`);
      fs.writeFileSync(pngPath, buffer);
      console.log(`✓ Converted to icon${size}.png`);
    });
    
    console.log('\n✓ All icons converted to PNG format!');
  } catch (err) {
    console.error('\nCanvas module not found. Install with: npm install canvas');
    console.log('Or convert SVG files manually using the methods above.');
  }
}
